### DISCUSSION TITLE
Question about sentence in docs

### DISCUSSION THREAD

**Discussion by axotion** [mikro-orm/mikro-orm#4211] [Q&A] [answered]
Hi,

In docs we have this line
>Keep in mind, that all handlers that are decorated with @UseRequestContext(), should NOT return anything.
https://mikro-orm.io/docs/usage-with-nestjs#request-scoped-handlers-in-queues

Why is that? What will happen if I return an entity from method with UseRequestContext?

E.g

```typescript


  async execute(withdrawUserItemCommand: WithdrawUserItemCommand) {
       const user = await this.getUser(withdrawUserItemCommand.userUuid);
       const modifierUser = await this.modifyUser(user);
}

  @UseRequestContext()
  private async getUser(userUuid: string): Promise<User> {
    return await this.userRepository.findOneByUuid(userUuid);
  }

  @UseRequestContext()
  private async modifyUser(user: User): Promise<User> {
    user.name  = 'aaa';
    await this.userRepository.persistAndFlush(user);
    return user;
  }

```

**Comment by B4nan** [comment]
You should not chain the calls, as it will always create new context. v6 renames the decorator to `@CreateRequestContext` so its clear, and adds a new one that will reuse existing context if available, which is suitable for such usage.

https://github.com/mikro-orm/mikro-orm/commit/68f357fc13927b4e4c668901240339d26db65f8b

**Reply by axotion** [reply]
But it is bad that it will always create new context? If I want clear em state in both getUser and modifyUser methods

**Reply by B4nan** [reply, marked as answer]
Yes, because the `user` property will hold detached entity, and such will produce insert query in the `modifyUser` method, which will result in duplicate PK. At least that is how it should behave.
